File: api/src/test/java/net/blueshell/api/factory/dto/SponsorDTOFactory.java
package net.blueshell.api.factory.dto;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.SponsorDTO;
import org.springframework.stereotype.Component;

/**
 * Factory for SponsorDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class SponsorDTOFactory extends BaseDtoFactory<SponsorDTO> {

    @Override
    public Class<SponsorDTO> targetType() {
        return SponsorDTO.class;
    }

    @Override
    public SponsorDTO createBasic() {
        SponsorDTO dto = new SponsorDTO();
        dto.setName(unique("Sponsor"));
        dto.setDescription("Test sponsor description");
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/SponsorDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.sponsor.dto.SponsorDTO
import org.springframework.stereotype.Component

/**
 * Factory for SponsorDTO test instances.
 */
@Component
class SponsorDTOFactory : BaseDtoFactory<SponsorDTO>() {

    override fun targetType(): Class<SponsorDTO> = SponsorDTO::class.java

    override fun createBasic(): SponsorDTO {
        val dto = SponsorDTO()
        dto.name = unique("Sponsor")
        dto.description = "Test sponsor description"
        return dto
    }
}
